pkg/collector/reconcile/configmap.go

```diff
@@ -124,6 +125,9 @@ func expectedConfigMaps(ctx context.Context, params Params, expected []corev1.Co
 		if err := params.Client.Patch(ctx, updated, patch); err != nil {
 			return fmt.Errorf("failed to apply changes: %w", err)
 		}
+		if configMapChanged(&desired, existing) {
+			params.Recorder.Event(updated, "Normal", "ConfigUpdate ", fmt.Sprintf("OpenTelemetry Config changed - %s/%s", desired.Namespace, desired.Name))
```

Comment:
Interesting, I haven't seen this one used before. We should indeed see if we should use this more often. Any docs on the usage recommendation?

Reply:
There is a mention on how to create events for resources created by controllers here - https://book-v1.book.kubebuilder.io/beyond_basics/creating_events.html.
This is still not documented in v2 book (https://book.kubebuilder.io/) - I guess its still pending.

I just figured it is the right way to handle our case as well as we have watch for configMap events already.